File: src/main/java/it/uniud/newbestsub/utils/Formula.java
package it.uniud.newbestsub.utils;

public class Formula {

    public static int stringComparison(String firstString, String secondString){
        int distance = 0;
        int i = 0;
        while (i < firstString.length()) {
            if (secondString.toCharArray()[i] != firstString.toCharArray()[i] ) {
                distance++;
            }
            i++;
        }
        return distance;
    }

    public static double getMean(double[] run, boolean[] useColumns){

        double mean = 0.0;

        int numberOfUsedCols = 0;
        for(int i=0; i<useColumns.length; i++){
            if(useColumns[i]){
                numberOfUsedCols++;
            }
        }

        for(int i=0; i<run.length;i++){
            if(useColumns[i]){
                mean = mean + run[i];
            }
        }
        mean = mean / ((double) numberOfUsedCols);

        return mean;
    }

    public static double getMean(double[] run, int[] useColumns){

        double mean = 0.0;

        int numberOfUsedCols = 0;
        for(int i=0; i<useColumns.length; i++){
            if(useColumns[i]==1){
                numberOfUsedCols++;
            }
        }

        for(int i=0; i<run.length;i++){
            if(useColumns[i]==1){
                mean = mean + run[i];
            }
        }
        mean = mean / ((double) numberOfUsedCols);

        return mean;
    }

}

File: src/main/java/it/uniud/newbestsub/utils/Formula.kt
package it.uniud.newbestsub.utils

object Formula {

    fun stringComparison(firstString: String, secondString: String): Int {
        var distance = 0
        var i = 0
        while (i < firstString.length) {
            if (secondString.toCharArray()[i] != firstString.toCharArray()[i]) {
                distance++
            }
            i++
        }
        return distance
    }

    fun getMean(run: DoubleArray, useColumns: BooleanArray): Double {

        var mean = 0.0

        var numberOfUsedCols = 0
        for (i in useColumns.indices) {
            if (useColumns[i]) {
                numberOfUsedCols++
            }
        }

        for (i in run.indices) {
            if (useColumns[i]) {
                mean = mean + run[i]
            }
        }
        mean = mean / numberOfUsedCols.toDouble()

        return mean
    }

    fun getMean(run: DoubleArray, useColumns: IntArray): Double {

        var mean = 0.0

        var numberOfUsedCols = 0
        for (i in useColumns.indices) {
            if (useColumns[i] == 1) {
                numberOfUsedCols++
            }
        }

        for (i in run.indices) {
            if (useColumns[i] == 1) {
                mean = mean + run[i]
            }
        }
        mean = mean / numberOfUsedCols.toDouble()

        return mean
    }

}
